### PR TITLE
Dev jacob

### PR DESCRIPTION
**Pull request by jacobrickman**
Added the Pathfinding System and Component, with placeholder basic/dumb pathfinding

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,23 +11,8 @@ file(GLOB_RECURSE GLSL "resources/*.glsl" "src/*.glsl")
 # Set the executable.
```
Comment (jaafersheriff) [inline CMakeLists.txt]: This is CMake file is specific to your build environment. Can you uncommit it and leave it uncommitted for future PRs?

File: CMakeLists_2.txt [hunk 2]
```diff
@@ -0,0 +1,100 @@
+cmake_minimum_required(VERSION 2.7)
```
Comment (jaafersheriff) [inline CMakeLists_2.txt]: ??

File: CMakeLists_git.txt [hunk 3]
```diff
@@ -0,0 +1,100 @@
+cmake_minimum_required(VERSION 2.7)
```
Comment (jaafersheriff) [inline CMakeLists_git.txt]: ???

File: resources/cube.obj [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Blender v2.71 (sub 0) OBJ File: ''
```
Comment (jaafersheriff) [inline resources/cube.obj]: Is this a necessary asset?

File: resources/cylinder.obj [hunk 5]
```diff
@@ -0,0 +1,231 @@
+# Blender v2.56 (sub 0) OBJ File: 'untitled.blend'
```
Comment (jaafersheriff) [inline resources/cylinder.obj]: Is this a necessary asset?

File: src/Engine/Component/CameraComponents/CameraController.cpp [hunk 6]
```diff
@@ -44,12 +46,32 @@ void CameraController::moveFront(float dt) {
     camera->setDirty();
 }
 
+void CameraController::strafeForward(float dt) {
```
Comment (jaafersheriff) [inline src/Engine/Component/CameraComponents/CameraController.cpp]: How is this different than moveFront()?

File: CMakeLists.txt [hunk 7]
```diff
@@ -11,23 +11,8 @@ file(GLOB_RECURSE GLSL "resources/*.glsl" "src/*.glsl")
...
 add_executable(${CMAKE_PROJECT_NAME} ${SOURCES} ${HEADERS} ${GLSL})
 
-# Visual Studio macro
```
Comment (Daskie) [inline CMakeLists.txt]: Why the changes to CMakeLists?

File: src/Engine/Util/Geometry.hpp [hunk 8]
```diff
@@ -114,15 +114,16 @@ struct Intersect {
...
         is(false),
         dist(std::numeric_limits<float>::infinity()),
         loc(),
         norm(),
         face(true)
     {}
 
-    Intersect::Intersect(
+    Intersect(
+        bool is,
```
Comment (Daskie) [inline src/Engine/Util/Geometry.hpp]: why did you add this field in? Would you ever construct an intersect with parameters that did not intersect?

File: src/Engine/Component/Component.hpp [hunk 9]
```diff
@@ -35,7 +35,9 @@ class Component {
         /* Receive a message sent by another component */
         //virtual void receiveMessage(Message &);
 
-    protected:
+
+
+    //protected:
```
Comment (Daskie) [inline src/Engine/Component/Component.hpp]: why did you comment this out?

Comment (jacobrickman) [reply src/Engine/Component/Component.hpp]: Needed access to the GameObject and kept getting errors, I should be able to use it in a class that implements Component right?

File: src/Engine/Util/Geometry.hpp (hunk 8 above)
Comment (jacobrickman) [reply src/Engine/Util/Geometry.hpp]: I didn't add any fields

Comment (jacobrickman) [reply src/Engine/Util/Geometry.hpp]: Should this constructor be different?
